app/controllers/cart_items_controller.rb

```diff
@@ -0,0 +1,57 @@
+class CartItemsController < ApplicationController
+  before_action :find_cart_id, only: %i(update destroy)
+
+  def create
+    @cart_item = current_cart.cart_items.new cart_item_params
+    if @cart_item.save
+      respond_to do |format|
+        session[:cart_id] = current_cart.id
+        format.html{redirect_to products_path}
+        format.js
+      end
+    else
+      format.html{redirect_to products_path}
+      format.js
+    end
+  end
+
+  def update
+    if @cart_item.update cart_item_params
+      respond_to do |format|
+        @cart_items = current_cart.cart_items
+        format.html{redirect_to carts_path}
+        format.js
+      end
+    else
+      format.html{redirect_to carts_path}
+      format.js
+    end
```

Comment:
```suggestion
    @error = "errors" unless @cart_item.update cart_item_params
    respond_to do |format|
        @cart_items = current_cart.cart_items if @error
        format.html{redirect_to carts_path}
        format.js
      end
```
Em thử xử lý như này xem có ok hơn không biến @error có thể báo lỗi ở view luôn